lib/ably/rest/channel.rb

```diff
@@ -40,15 +40,14 @@ def initialize(client, name, channel_options = {})
         @push    = PushChannel.new(self)
       end
 
-      # Publish one or more messages to the channel.
-      #
-      # @param name [String, Array<Ably::Models::Message|Hash>, nil]   The event name of the message to publish, or an Array of [Ably::Model::Message] objects or [Hash] objects with +:name+ and +:data+ pairs
-      # @param data [String, ByteArray, nil]   The message payload unless an Array of [Ably::Model::Message] objects passed in the first argument
-      # @param attributes [Hash, nil]   Optional additional message attributes such as :extras, :id, :client_id or :connection_id, applied when name attribute is nil or a string
+      # Publish one or more messages to the channel. Three overloaded forms
+      # @param name [String, Array<Ably::Models::Message|Hash>, Ably::Models::Message, nil]   The event name of the message to publish, or an Array of [Ably::Model::Message] objects or [Hash] objects with +:name+ and +:data+ pairs, or a single Ably::Model::Message object
+      # @param data [String, ByteArray, Hash, nil]   The message payload unless an Array of [Ably::Model::Message] objects passed in the first argument, in which case an optional hash of query parameters
+      # @param attributes [Hash, nil]   Optional additional message attributes such as :extras, :id, :client_id or :connection_id, applied when name attribute is nil or a string (Deprecated, will be removed in 1.2 in favour of constructing a Message object)
```

Comment:
Why would this be removed in 1.2?  That's a breaking change and unnecessary.

Reply:
@mattheworiordan Sorry, that's my mistake - it should say 2.0, as it's a breaking change. I'll fix that.

The spec item that requires it be deprecated is https://docs.ably.io/client-lib-development-guide/features/#RSL1h . The docs pr for that was https://github.com/ably/docs/pull/742 , and the issues it addressed were https://github.com/ably/docs/issues/644 and https://github.com/ably/docs/issues/737#issuecomment-538471207 , those issues have all the discussion that lead to us agreeing to deprecate messageAttributes as a method parameter of publish().